Assets/HoloToolkit/SpectatorView/Scripts/Networking/NewDeviceDiscovery.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace HoloToolkit.Unity.SpectatorView
+{
+    /// <summary>
+    /// In this instance the HoloLens will be listening to broadcasts from the mobile device to turn on the camera
+    /// </summary>
+    public class NewDeviceDiscovery : NetworkDiscovery
+    {
+        /// <summary>
+        /// Is the device a host or a client? (HoloLens or mobile?)
+        /// </summary>
+        private bool isHost;
+
+        /// <summary>
+        /// Component used to detect a AR marker from the HoloLens
+        /// </summary>
+        [SerializeField]
+        [Tooltip("Component used to detect a AR marker from the HoloLens")]
+        private MarkerDetectionHololens markerDetectionHololens;
+
+        /// <summary>
+        /// Component used to detect a AR marker from the HoloLens
+        /// </summary>
+        public MarkerDetectionHololens MarkerDetectionHololens
+        {
+            get { return markerDetectionHololens; }
+            set { markerDetectionHololens = value; }
+        }
+
+        private void Awake()
+        {
+#if NETFX_CORE
+            try
+            {
+			    OpenCVUtils.CheckOpenCVWrapperHasLoaded();
+            }
+            catch (Exception e)
+            {
+				Debug.LogError(e);
+				gameObject.SetActive(false);
+                return;
+            }
+#endif
+            isHost = FindObjectOfType<PlatformSwitcher>().TargetPlatform == PlatformSwitcher.Platform.Hololens;
+            // The client doesn't have to wait for the server to be started, but this works best if the component
+            // waits for the remaining networking bits to have warmed up,
+            // just give it a couple of seconds and then start it
+            if (!isHost)
+            {
+                Invoke("ManualStart", 4f);
```

Comment:
Not a fan of invoking a method via string.

Making Manual Start an IEnumerator and adding `Yield new WaitForSeconds(4f);` would be a valid fix.

Reply:
this is arguably easier to read / less code